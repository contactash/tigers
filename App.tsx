import { StatusBar } from 'expo-status-bar';
import React, {useCallback, useEffect, useState} from 'react';
import {Button, StyleSheet, Text, View} from 'react-native';

type CustomTextProps = {
  toggleText: boolean;
  color?: string;
}

function CustomText({ toggleText, color='green' }: CustomTextProps) {
  return toggleText ?
      <Text  testID={'myText'}  style={{color}}>Open up App.tsx to start working on your app!</Text> :
      <Text  testID={'myText'}  style={{color}}>Open up App.!</Text>
}

export default function App() {

  const [toggleText, setToggleText] = useState(false);
  const [color, setColor] = useState(false);

  function getOnPress() {
    return () => setToggleText(!toggleText);
  }

  return (
    <View style={styles.container}>
      <CustomText toggleText={toggleText} color={'blue'}/>
      <View style={styles.box1} />
      <View style={styles.box2} />
      <StatusBar style="auto"/>
      <Button testID={'myButton'} title={'Press Here'} onPress={getOnPress()}/>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  box1: {
    width: 50,
    height: 50,
    backgroundColor: 'blue'
  },
  box2: {
    width: 50,
    height: 50,
    backgroundColor: 'red'
  },
});


