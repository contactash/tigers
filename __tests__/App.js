import 'react-native'
import React from 'react'
import App from '../App.tsx'
import {create , act} from 'react-test-renderer'

describe('App', () => {
    const testRenderer = create(<App />);
    const tree = testRenderer.toJSON();
    const testInstance = testRenderer.root;
    it('should render correctly', () => {
        expect(tree).toMatchSnapshot()
        expect(tree.children.length).toBe(4)
    })
    
    it('should toggle the test on button press', () => {
        const textComponent = testInstance.findByProps({testID: 'myText'})
        expect(textComponent.props.children).toEqual('Open up App.!');
        
        const button = testInstance.findByProps({testID: 'myButton'}).props;
        
        act(() => {
            button.onPress();
        });
        expect(textComponent.props.children).toEqual('Open up App.tsx to start working on your app!');
    })
})
