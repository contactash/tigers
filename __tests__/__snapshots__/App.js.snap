// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "blue",
      }
    }
    testID="myText"
  >
    Open up App.!
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "blue",
        "height": 50,
        "width": 50,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "red",
        "height": 50,
        "width": 50,
      }
    }
  />
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="myButton"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press Here
      </Text>
    </View>
  </View>
</View>
`;
